src/core/qgsabstractcontentcache.h

```diff
@@ -367,14 +369,31 @@ class CORE_EXPORT QgsAbstractContentCache : public QgsAbstractContentCacheBase
         QMetaObject::invokeMethod( const_cast< QgsAbstractContentCacheBase * >( qobject_cast< const QgsAbstractContentCacheBase * >( this ) ), "onRemoteContentFetched", Qt::QueuedConnection, Q_ARG( QString, path ), Q_ARG( bool, true ) );
       } );
 
-      QgsApplication::taskManager()->addTask( task );
+      if ( synchrone )
+      {
+        if ( task->run() )
+        {
+          if ( mRemoteContentCache.contains( path ) )
+          {
+            // We got the file!
+            return *mRemoteContentCache[ path ];
```

Comment:
`task` is leaked here

Reply:
do I have to add `task->deleteLater()` or `delete task` ?

Reply:
Done